backend/lucene/src/main/java/org/hibernate/search/backend/lucene/work/impl/LuceneSearchWork.java

```diff
@@ -89,7 +89,7 @@ else if ( limit == 0L ) {
 			return 0;
 		}
 		else {
-			return Math.min( (int) ( offset + limit ), reader.maxDoc() );
```

Comment:
This cast should indeed be removed, but the new code doesn't follow the Hibernate code style. There should be a space after an opening parenthesis (missing here), and a space before a closing parenthesis.

See http://hibernate.org/search/contribute/, in particular see the code style configuration for [Eclipse](https://github.com/hibernate/hibernate-ide-codestyles/blob/master/eclipse/hibernate-code-template-search.xml) or [IntelliJ IDEA](https://github.com/hibernate/hibernate-ide-codestyles/blob/master/intellij-14/hibernate_noorm.xml).